sources/core/objects/Floor.cpp

```diff
@@ -28,7 +28,6 @@ void Floor::AfterWorldCreation()
 {
     ITurf::AfterWorldCreation();
     SetOpen(open_);
-    SetState(floor_type_);
```

Comment:
You are right. I forgot that `SetOpen` is called and that it sets states. So everything should work fine now.